cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -507,9 +508,12 @@ public void pullPackage(String org, String name, String version, Path packagePat
                 Optional<String> balaFileName = Optional.ofNullable(packagePullResponse.header(CONTENT_DISPOSITION));
                 Optional<String> deprecationFlag = Optional.ofNullable(packagePullResponse.header(IS_DEPRECATED));
                 Optional<String> deprecationMsg = Optional.ofNullable(packagePullResponse.header(DEPRECATE_MESSAGE));
+                Optional<String> digest = Optional.ofNullable(packagePullResponse.header(DIGEST));
 
+                String digestVal = digest.isPresent() ? digest.get() : "";
```

Comment:
IMO, it's better to have the Optional<trueDigest> rather than assigning an empty string if it is empty. In L465 we can add the null check before validating.